src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -22,43 +22,84 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
 
         if (!paymentMethod.initializationData.clientId) {
             throw new InvalidArgumentError();
         }
 
         return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
             .then(state => {
-                const config = state.config.getStoreConfig();
                 const cart = state.cart.getCart();
 
-                if (!config) {
-                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
-                }
-
                 if (!cart) {
                     throw new MissingDataError(MissingDataErrorType.MissingCart);
                 }
 
-                const paramsScript = {
+                const buttonParams: ButtonsOptions = {
+                    createOrder: () => this._setupPayment(cart),
+                    onApprove: data => this._tokenizePayment(paymentMethod.id, data),
+                };
+
+                if (paypalOptions && paypalOptions.style) {
+                    buttonParams.style = this.validateStyleParams(paypalOptions.style);
+                }
+
+                const paramsScript: PaypalCommerceScriptOptions = {
                     clientId: paymentMethod.initializationData.clientId,
-                    currency: config.currency.code,
+                    currency: cart.currency.code,
+                    commit: false,
+                    intent: paymentMethod.initializationData.intent,
                 };
 
+                if (!paymentMethod.initializationData.isPayPalCreditAvailable) {
+                    paramsScript.disableFunding = 'credit';
+                }
+
                 return this._paypalScriptLoader.loadPaypalCommerce(paramsScript)
-                    .then(paypal => {
-                        return paypal.Buttons({
-                            createOrder: () => this._setupPayment(cart),
-                            onApprove: data => this._tokenizePayment(paymentMethod.id, data),
-                        }).render(`#${options.containerId}`);
-                    });
+                    .then(paypal => paypal.Buttons(buttonParams).render(`#${options.containerId}`));
             });
     }
 
     deinitialize(): Promise<void> {
         return Promise.resolve();
     }
 
+    validateStyleParams(style: PaypalButtonStyleOptions): PaypalButtonStyleOptions {
+        const updatedStyle: PaypalButtonStyleOptions = { ...style };
+        const { label, color, layout, shape, height, tagline } = style;
+
+        if (label && !StyleButtonLabel[label]) {
+            delete updatedStyle.label;
+        }
+
+        if (layout && !StyleButtonLayout[layout]) {
+            delete updatedStyle.layout;
+        }
+
+        if (color && !StyleButtonColor[color]) {
+            delete updatedStyle.color;
+        }
+
+        if (shape && !StyleButtonShape[shape]) {
+            delete updatedStyle.shape;
+        }
+
+        if (typeof height === 'number') {
+            updatedStyle.height = height < 25
```

Comment:
you could do something like  
```js
updatedStyle.height = Math.min(Math.max(height, 55), 25);
```

Reply:
That's great idea. Thank you!
But tslint doesn't understand value
Can I write `type height = 25...55; updatedStyle.height = Math.min(Math.max(height, 25), 55) as height;` ?

I'm not sure, that's valid

Reply:
you are right it's a bit annoying that TSlint doesn't recognize it, you could keep it as you had it then to avoid overriding the type.